histoqc/ClassificationModule.py

```diff
@@ -202,7 +202,10 @@ def byExampleWithFeatures(s, params):
     img = s.getImgThumb(s["image_work_size"])
     feats = compute_features(img, params)
     cal = clf.predict_proba(feats.reshape(-1, feats.shape[2]))
-    cal = cal.reshape(img.shape[0], img.shape[1], 2)
+    # get the classes #
+    cal_classes = cal.shape[1]
+    # reshape by predict classes
+    cal = cal.reshape(img.shape[0], img.shape[1], cal_classes)
 
     mask = cal[:, :, 1] > thresh
```

Comment:
i'm not entirely sure i understand this code here. if you have say 4 classes, won't think always produce a binary mask based on the 1st channel because of line 210? 

```
mask = cal[:, :, 1] > thresh
```

is this generalizable to multiple classes, and/or what is the situation where we have more than 2 classes? how does the user specify that which channel they want to be the defining class? 

it seems like we may only want to support 2 channel masks to keep life simple, and rely on the user to provide the correct type of input?

Reply:
Hi @choosehappy,
It is relevant to #250 and #254. My explanation is on #254.

Reply:
left a comment there. i still don't understand why this is needed or why we're not always doing binary classification? shouldn't need any multi-class stuff